diesel/src/pg/types/array.rs

```diff
@@ -134,10 +134,13 @@ impl<'a, ST, T> ToSql<Array<ST>, Pg> for &'a [T] where
         let mut buffer = Vec::new();
         for elem in self.iter() {
             let is_null = try!(elem.to_sql(&mut buffer));
-            assert!(is_null == IsNull::No, "Arrays containing null are not supported");
-            try!(out.write_i32::<NetworkEndian>(buffer.len() as i32));
-            try!(out.write_all(&buffer));
-            buffer.clear();
+            if let IsNull::No = is_null {
+                try!(out.write_i32::<NetworkEndian>(buffer.len() as i32));
+                try!(out.write_all(&buffer));
+                buffer.clear();
+            } else {
+                try!(out.write_i32::<NetworkEndian>(-1));
```

Comment:
I can link to the line in the source code? There is no documentation on this. But we also have https://github.com/diesel-rs/diesel/blob/3d985d717b0caaad931464e8a658c1f73bc9a57e/diesel/src/pg/types/array.rs#L54-L56 ;)